src/python/pants/backend/python/python_chroot.py

```diff
@@ -249,15 +249,16 @@ def _resolve_multi(self, requirements, find_links):
     context = self._python_repos.get_network_context()
 
     for platform in platforms:
-      requirements_cache_dir = os.path.join(self._python_setup.resolver_cache_dir, str(self._interpreter.identity))
+      requirements_cache_dir = os.path.join(self._python_setup.resolver_cache_dir,
+                                            str(self._interpreter.identity))
       distributions[platform] = resolve(
         requirements=[req.requirement for req in requirements],
         interpreter=self._interpreter,
         fetchers=fetchers,
         platform=platform,
         context=context,
         cache=requirements_cache_dir,
-        cache_ttl=self._python_setup.resolver_cache_ttl,
+        cache_ttl=10 * 365 * 86400,
```

Comment:
Would you be comfortable with this being parameterized when resolving user code (i.e., here and in `pex_build_util.py`) but not when resolving pants plugins themselves? That is supposed to be a one-time thing, and is a known, fixed universe, so the risk seems low?  Or we could add a global option for this somewhere else. What I don't want is to have `plugin_resolver.py` reference PythonSetup (see below for more details on this).